messages.rb

```diff
@@ -505,6 +506,84 @@ def send_diff(event)
   event << "Score changes on #{current.format_name} since #{since}:\n```#{diff}```"
 end
 
+def do_analysis(scores, rank)
+  run = scores.download_scores(rank)
+  if !run.nil?
+    player = run['user_name']
+    replay_id = run['replay_id'].to_s
+    score = "%.3f" % [run['score'].to_f / 1000]
+    analysis = scores.analyze_replay(replay_id)
+    gold = "%.0f" % [((run['score'].to_f / 1000) + (analysis.size.to_f / 60) - 90) / 2]
+    result = {'player' => player, 'scores' => scores.format_name, 'rank' => rank, 'score' => score, 'analysis' => analysis, 'gold' => gold}
+    return result
+  else
+    return nil
+  end
+end
+
+def send_analysis(event)
+  msg = event.content
+  scores = parse_level_or_episode(msg)
+  ranks = msg.scan(/\s+([0-9][0-9]?)/).map{ |r| r[0].to_i }.reject{ |r| r < 0 || r > 19 }
```

Comment:
Super minor, but this seems like it's begging to be broken out into a `parse_ranks` function (I know we already have `parse_rank`, so you might need to consider naming and come up with something a little better, but we're generally doing that 'parsing' work separately from the 'formatting' work).

Reply:
(and the line 528 below)

Reply:
Good idea, also `parse_ranks` is the most intuitive name I can think of.